### PR TITLE
Improving Taylor Diagram Dialog

### PR DESCRIPTION
**Pull request by N-thony**
Improved the design of the Taylor Diagram dialog and small changes in the code. This fixes the comments in issues #5885 
@africanmathsinitiative/developers this is ready to review. Thanks.

### CODE REVIEW

**Comment by lloyddewit** [comment]
@shadrackkibet please could you review? thanks

File: instat/dlgTaylorDiagram.Designer.vb [hunk 1]
```diff
@@ -42,10 +42,10 @@ Partial Class dlgTaylorDiagram
         resources.ApplyResources(Me.lblObserved, "lblObserved")
         Me.lblObserved.Name = "lblObserved"
         '
-        'lblEstimate
+        'lblEstimated
         '
-        resources.ApplyResources(Me.lblEstimate, "lblEstimate")
-        Me.lblEstimate.Name = "lblEstimate"
+        resources.ApplyResources(Me.lblEstimated, "lblEstimated")
+        Me.lblEstimated.Name = "lblEstimated"
```
Comment (shadrackkibet) [inline instat/dlgTaylorDiagram.Designer.vb:48]: This label seems to have disappeared.

Comment (shadrackkibet) [reply instat/dlgTaylorDiagram.Designer.vb:48]: This is what I get.

image

**Comment by shadrackkibet** [comment]
Please change the current receiver to be the observed receiver. This is because the station receiver is auto-filled.

File: instat/dlgTaylorDiagram.Designer.vb (hunk 1 above)
Comment (N-thony) [reply instat/dlgTaylorDiagram.Designer.vb:48]: This is different from what I have but let me push again.

**Comment by N-thony** [comment]
@shadrackkibet I have pushed my branch with small changes about the design requested in issue #5885

**Comment by shadrackkibet** [comment]
What about the button to the sub dialog?

File: instat/dlgTaylorDiagram.vb [hunk 2]
```diff
@@ -40,30 +40,30 @@ Public Class dlgTaylorDiagram
...
         ucrReceiverSingleObserved.SetIncludedDataTypes({"numeric"})
 
-        ucrReceiverSingleEstimate.SetParameter(New RParameter("mod", 1))
-        ucrReceiverSingleEstimate.Selector = ucrSelectorTaylorDiagram
-        ucrReceiverSingleEstimate.SetParameterIsString()
-        ucrReceiverSingleEstimate.SetIncludedDataTypes({"numeric"})
+        ucrReceiverSingleEstimated.SetParameter(New RParameter("mod", 2))
+        ucrReceiverSingleEstimated.Selector = ucrSelectorTaylorDiagram
+        ucrReceiverSingleEstimated.SetParameterIsString()
+        ucrReceiverSingleEstimated.SetIncludedDataTypes({"numeric"})
```
Comment (shadrackkibet) [inline instat/dlgTaylorDiagram.vb:51]: Change selector header here i.e ` ucrReceiverSingleEstimated.strSelectorHeading = "Numerics"`. Also do the same for other receivers where variables are restricted.

File: instat/dlgTaylorDiagram.vb [hunk 3]
```diff
@@ -40,30 +40,30 @@ Public Class dlgTaylorDiagram
...
-        ucrReceiverSingleStation.SetParameterIsString()
-        ucrReceiverSingleStation.bExcludeFromSelector = True
-        ucrReceiverSingleStation.SetClimaticType("station")
-        ucrReceiverSingleStation.bAutoFill = True
-
-        ucrReceiverSingleObserved.SetParameter(New RParameter("obs", 0))
+        ucrReceiverSingleObserved.SetParameter(New RParameter("obs", 1))
         ucrReceiverSingleObserved.Selector = ucrSelectorTaylorDiagram
         ucrReceiverSingleObserved.SetParameterIsString()
         ucrReceiverSingleObserved.SetIncludedDataTypes({"numeric"})
```
Comment (shadrackkibet) [inline instat/dlgTaylorDiagram.vb:46]: Also here `ucrReceiverSingleObserved.strSelectorHeading = "Numerics"`

**Comment by N-thony** [comment]
> What about the button to the sub dialog?

I have added the button and changed the current receiver to be the observed receiver.

File: instat/dlgTaylorDiagram.vb [hunk 4]
```diff
@@ -40,30 +40,30 @@ Public Class dlgTaylorDiagram
...
+        ucrReceiverSingleEstimated.Selector = ucrSelectorTaylorDiagram
+        ucrReceiverSingleEstimated.SetParameterIsString()
+        ucrReceiverSingleEstimated.SetIncludedDataTypes({"numeric"})
 
-        ucrReceiverMultipleGroup.SetParameter(New RParameter("group", 2))
+        ucrReceiverMultipleGroup.SetParameter(New RParameter("group", 3))
         ucrReceiverMultipleGroup.Selector = ucrSelectorTaylorDiagram
         ucrReceiverMultipleGroup.SetParameterIsString()
         ucrReceiverMultipleGroup.SetIncludedDataTypes({"factor", "character"})
         ucrReceiverMultipleGroup.bExcludeFromSelector = True
```
Comment (shadrackkibet) [inline instat/dlgTaylorDiagram.vb:57]: Also here `  ucrReceiverMultipleGroup.strSelectorHeading = "Factors & Characters"`

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

**Comment by shadrackkibet** [comment]
That is all for now. Once this is done @lloyddewit please proceed with the next steps.

**Comment by N-thony** [comment]
@lloyddewit I solved the comments above about the small change in the code. This is now ready for the next steps. Thanks.

**Comment by lloyddewit** [comment]
@rdstern please could you test? thanks

**Review by shadrackkibet — APPROVED** [review]
Looks good. Over to you @rdstern @dannyparsons @lloyddewit

**Comment by shadrackkibet** [comment]
I am now taking over this PR.

**Comment by lloyddewit** [comment]
> 
> 
> I am now taking over this PR.

@shadrackkibet Thanks, I assume that you intend to make additional changes. Should we ait for these changes before we test/review?

**Comment by shadrackkibet** [comment]
I now have a new PR (#6339) branched from this one, with additional changes and it is ready for review.

**Comment by lloyddewit** [comment]
> 
> 
> I now have a new PR (#6339) branched from this one, with additional changes and it is ready for review.

Thank you for the explanation. Sorry, I should have looked at PR #6339 before I commented.

I'm not sure what to do next for this PR, it depends on your intention for PR #6339.
Should we:
- merge this PR now?
- close after PR #6339 merged?
- wait for this PR to merge automatically when PR #6339 merged?

Thanks for your advice

**Comment by shadrackkibet** [comment]
Wait for this PR to merge automatically when PR #6339 is merged.

**Pull request merged**